### ISSUE TITLE
Determine why `RefName` is of `Sentence` type

### ISSUE THREAD

**Issue by niazim3** [JacquesCarette/Drasil#551]
As mentioned in comments of commit 7021dd2, the type of `ShortNm` maybe should be `= ShortName String` instead of `ShortName Sentence`; before doing so, however, it would be beneficial to see how `RefName` (the hack `ShortNm` is replacing) is used (i.e. if any `Sentence` features are used that cannot be done via `String`).

**Comment by niazim3** [comment]
### `RefName`
- Wherever it is used in the Language subfolder doesn't seem to be using features of `Sentence`
![image](https://user-images.githubusercontent.com/28247301/40492660-fcb35b80-5f3e-11e8-8ff6-482fbcf05ef8.png)

### `Ref` uses `RefName` to build a `Sentence` type
- https://github.com/JacquesCarette/Drasil/blob/8e924ffbb30e5c54ec62bf2f2bcdb18541adf8fa/code/Language/Drasil/Spec.hs#L22-L30
- https://github.com/JacquesCarette/Drasil/blob/8e924ffbb30e5c54ec62bf2f2bcdb18541adf8fa/code/Language/Drasil/Printing/AST.hs#L38-L44
- This can easily be converted to a sentence since this `spec` function requires it https://github.com/JacquesCarette/Drasil/blob/8e924ffbb30e5c54ec62bf2f2bcdb18541adf8fa/code/Language/Drasil/Printing/Import.hs#L243
- `Sentence` and `Ref` get treated the same here
https://github.com/JacquesCarette/Drasil/blob/8e924ffbb30e5c54ec62bf2f2bcdb18541adf8fa/code/Language/Drasil/TeX/Print.hs#L225-L232
- `spec` doesn't care what type `RefName` has here:
https://github.com/JacquesCarette/Drasil/blob/8e924ffbb30e5c54ec62bf2f2bcdb18541adf8fa/code/Language/Drasil/TeX/Print.hs#L248-L256

In short, it doesn't look like having `RefName` as a `Sentence` type is serving a purpose that it as a `String` type cannot.

**Comment by niazim3** [comment]
After going through some of the implementation, it became apparent that one instance that poses an issue wrt turning `RefName` into a `String`, is with the following: 
```
-- | Sections have a title ('Sentence') and a list of contents ('SecCons')
data Section = Section Title [SecCons] RefAdd

instance HasShortName Section where shortname (Section t _ _) = t
```
, where `Title` is a `Sentence` type and expects a `Sentence` to create a `CitationField`:
https://github.com/JacquesCarette/Drasil/blob/5f322c79259e27c26f6f7a65b6bc2344aa35b83f/code/Language/Drasil/Chunk/Citation.hs#L50.
Although @szymczdm mentioned that it's possible that changing it (and all the other `CitationField` inputs) to a `String` won't be incorrect, currently looking into a way that allows a `String` to be taken in and automatically converts it to a `Sentence`, so as not to deviate from this task at hand.

**Comment by JacquesCarette** [comment]
Actually, I think it is a 'mistake' to automatically take the section title to be the shortname. I think sections should have user-defined short names (perhaps optional?). These can then be used for `shortname` (and thus obey the rules of `shortname`.  That is, de facto, what both HTML and LaTeX do anyways (i.e. the label or id associated to a section is separate from the text of the section, and has different rules).  So it makes most sense to follow their lead.

**Comment by niazim3** [comment]
In that case, `Section` should still be an instance of the `HasShortName` class, but the title should not be associated with it, correct?
Instead, `Section`'s `shortname` will get what then? A `Sentence` that's passed in as a a `Title` by a user?

**Comment by JacquesCarette** [comment]
Correct. `Section` should get an additional field, a `String`, which is its `shortname`. So creators of a `Section` will have to pass in an extra amount of content.

**Comment by smiths** [comment]
I agree with @JacquesCarette.  The `shortname` should not be automatically constructed.  The user can think of a name, just like they do for labels in LaTeX.  We have experimented with automatically generated names in Drasil in the past and they seem to be more confusing than helpful, at least to me.
